<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Carlos Marrero PÃ©rez" />

<meta name="date" content="2021-05-05" />

<title>PRJNA693328</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">PRJNA693328</h1>
<h4 class="author">Carlos Marrero PÃ©rez</h4>
<h4 class="date">2021-05-05</h4>



<div id="tarea-2" class="section level2">
<h2>Tarea 2</h2>
<div id="elecciÃ³n-del-experimento" class="section level3">
<h3>ElecciÃ³n del experimento</h3>
<ul>
<li><p>Se ha seleccionado el estudio <strong>PRJNA693328</strong>.</p></li>
<li><p><strong>TÃ­tulo</strong>: Abundancia de las cÃ©lulas T CD8+ sobrerrepresentada en infiltrados de tumores humanos.</p></li>
</ul>
<p>Varias formas de inmunoterapia consiguen restaurar las respuestas inmunitarias mediadas por cÃ©lulas T, pero solo en algunos pacientes y tipos de cÃ¡ncer. Se han tomado muestras CD8+ de diversos tumores (de pulmÃ³n y colorrectal). Se observa que esas diferencias de respuesta a inmunoterapia estÃ¡ mediada por la expresiÃ³n de CD39, tipificado como CD39- (si no ocurre) y CD39+ (si existe). Las frecuencias de expresiÃ³n de CD39 correlacionan con parÃ¡metros clÃ­nicos importantes, es decir, se puede utilizar mediciones de la expresiÃ³n de CD39 como manera de cuantificar o aislar cÃ©lulas T con efecto <em>bystander</em>, relacionadas con la exacerbaciÃ³n de sÃ­ntomas.</p>
</div>
</div>
<div id="con-un-bule-en-r-descargamos-todos-los-datos-en-formato-.sra" class="section level2">
<h2>Con un bule en R descargamos todos los datos en formato .SRA</h2>
<pre eval="doAll"><code>options(timeout=10000)
for (i in 36:43) {
  url = paste0(&quot;https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos2/sra-pub-run-13/SRR70591&quot;,i,&quot;/SRR70591&quot;,i,&quot;.1&quot;)
  download.file(url,destfile = paste0(&quot;../data/SRR70591&quot;,i,&quot;.fastq.gz&quot;), downloadMethod=&quot;wget&quot;)
} 
setwd(&quot;../data&quot;)</code></pre>
</div>
<div id="descargamos-el-genoma-para-star" class="section level2">
<h2>Descargamos el genoma para STAR</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/GRCh38.p13.genome.fa.gz</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gzip</span> <span class="at">-d</span> GRCh38.p13.genome.fa.gz</span></code></pre></div>
</div>
<div id="alineamiento-con-star" class="section level2">
<h2>Alineamiento con STAR</h2>
<p>Generamos el fichero de Ã­ndices (aunque podrÃ­amos haber descargado la versiÃ³n indexada directamente) con STAR. A continuaciÃ³n, alineamos con STAR, creamos los .bam con SAMtools y los ordenamos. A continuaciÃ³n, eliminamos todos los datos generados excepto los .bam ordenados.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">STAR</span> <span class="at">--runMode</span> genomeGenerate <span class="at">--runThreadN</span> 6 <span class="at">--genomeDir</span> genomeDir0 <span class="at">--genomeFastaFiles</span> GRCh38.p13.genome.fa</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> variable <span class="kw">in</span> <span class="dt">{</span><span class="dv">36</span><span class="dt">..</span><span class="dv">43</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> <span class="ex">fastq-dump</span> <span class="at">-I</span> <span class="at">--split-files</span> SRR70591<span class="va">$variable</span>.sra<span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="bu">let</span> <span class="va">variable2</span><span class="op">=</span><span class="dv">37</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> variable <span class="kw">in</span> <span class="dt">{</span><span class="dv">36</span><span class="dt">..</span><span class="dv">43</span><span class="dt">}</span><span class="kw">;</span><span class="cf">do</span> <span class="ex">STAR</span> <span class="at">--genomeDir</span> ./ <span class="at">--runThreadN</span> 14 starIndex <span class="at">--readFilesIn</span> SRR70591<span class="va">${variable}</span>_1.fastq SRR70591<span class="va">${variable}</span>_2.fastq <span class="at">--outFileNamePrefix</span> paired<span class="va">${variable}</span><span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> variable <span class="kw">in</span> <span class="dt">{</span><span class="dv">36</span><span class="dt">..</span><span class="dv">43</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> <span class="ex">samtools</span> view <span class="at">-S</span> <span class="at">-b</span> paired<span class="va">${variable}</span>Aligned.out.sam <span class="op">&gt;</span> paired<span class="va">${variable}</span>Aligned.bam<span class="kw">;</span><span class="cf">done</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> variable <span class="kw">in</span> <span class="dt">{</span><span class="dv">36</span><span class="dt">..</span><span class="dv">43</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> <span class="ex">samtools</span> sort paired<span class="va">${variable}</span>Aligned.bam <span class="at">-o</span> paired<span class="va">${variable}</span>Aligned_sort.bam<span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Eliminamos todo lo que no sean nuestros .bam ordenados</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="pp">*</span>.sam <span class="pp">*</span>.fastq SAindex SA <span class="pp">*</span>.txt <span class="pp">*</span>.out <span class="pp">*</span>.tab <span class="pp">*</span>.fa datalist</span></code></pre></div>
</div>
<div id="creamos-la-lista-de-.bam-y-descarga-del-.gtf" class="section level2">
<h2>Creamos la lista de .bam y descarga del .gtf</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Lista de los ficheros .bam</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="pp">*</span>.bam <span class="op">&gt;</span> bamfiles.txt</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># El fichero de anotaciÃ³n</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_37/gencode.v37.annotation.gtf.gz</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">gzip</span> <span class="at">-d</span> gencode.v37.annotation.gtf.gz</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="pp">*</span>.gz</span></code></pre></div>
<div id="usando-rsamtools-para-el-conteo-de-las-reads-alineadas" class="section level3">
<h3>Usando Rsamtools para el conteo de las reads alineadas</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rsamtools)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicFeatures)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicAlignments)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>sampleTable <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;bamfiles.txt&quot;</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>dirActualData <span class="ot">=</span>  <span class="fu">paste</span>(<span class="fu">getwd</span>(),<span class="st">&quot;/&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>fls <span class="ot">=</span> <span class="fu">paste</span>(dirActualData,sampleTable[,<span class="dv">1</span>],<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>bamLst <span class="ot">=</span> <span class="fu">BamFileList</span>(fls, <span class="at">index=</span><span class="fu">character</span>(),<span class="at">yieldSize=</span><span class="dv">100000</span>,<span class="at">obeyQname=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>gtfFile <span class="ot">=</span> <span class="st">&quot;gencode.v37.annotation.gtf&quot;</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">=</span> <span class="fu">makeTxDbFromGFF</span>(gtfFile, <span class="at">format=</span><span class="st">&quot;gtf&quot;</span>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">=</span> <span class="fu">exonsBy</span>(txdb, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>PRJNA693328 <span class="ot">=</span> <span class="fu">summarizeOverlaps</span>(<span class="at">features =</span> genes, <span class="at">read=</span>bamLst,</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>                                <span class="at">mode=</span><span class="st">&quot;Union&quot;</span>,</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>                                <span class="at">singleEnd=</span><span class="cn">FALSE</span>, <span class="do">## Son lecturas apareadas</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>                                <span class="at">ignore.strand=</span><span class="cn">TRUE</span>,</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>                                <span class="at">fragments=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="metadatos" class="section level3">
<h3>Metadatos</h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>SampleName <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;SAMN08975667&quot;</span>, <span class="st">&quot;SAMN08975666&quot;</span>, <span class="st">&quot;SAMN08975665&quot;</span>, <span class="st">&quot;SAMN08975664&quot;</span>,<span class="st">&quot;SAMN08975663&quot;</span>,<span class="st">&quot;SAMN08975662&quot;</span>, <span class="st">&quot;SAMN08975661&quot;</span>, <span class="st">&quot;SAMN08975660&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>Run <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;SRR7059136&quot;</span>, <span class="st">&quot;SRR7059137&quot;</span>, <span class="st">&quot;SRR7059138&quot;</span>, <span class="st">&quot;SRR7059139&quot;</span>, <span class="st">&quot;SRR7059140&quot;</span>, <span class="st">&quot;SRR7059141&quot;</span>, <span class="st">&quot;SRR7059142&quot;</span>, <span class="st">&quot;SRR7059143&quot;</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>CD8_Status<span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>CD8_Status <span class="ot">=</span> <span class="fu">factor</span>(CD8_Status,<span class="at">levels=</span><span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;CD39-&quot;</span>, <span class="st">&quot;CD39+&quot;</span>))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(PRJNA693328) <span class="ot">=</span> <span class="fu">DataFrame</span>(SampleName, Run, CD8_Status)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(PRJNA693328)</span></code></pre></div>
</div>
<div id="metadatos-experimentales" class="section level3">
<h3>Metadatos experimentales</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata</span>(PRJNA693328)<span class="ot">=</span><span class="fu">list</span>(<span class="st">&quot;Experimenter name&quot;</span><span class="ot">=</span><span class="st">&quot;Kaibo Duan&quot;</span>,</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Contact&quot;</span><span class="ot">=</span><span class="st">&quot;duan_kaibo@immunol.a-star.edu.sg&quot;</span>,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Organization&quot;</span><span class="ot">=</span><span class="st">&quot;Singapore Immunology Network&quot;</span>,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Title&quot;</span><span class="ot">=</span><span class="st">&quot;Bystander CD8 + T cells are abundant and phenotypically distinct in human tumour infiltrates&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;URL&quot;</span><span class="ot">=</span><span class="st">&quot;https://pubmed.ncbi.nlm.nih.gov/29769722/&quot;</span>, </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;Abstract&quot;</span><span class="ot">=</span> <span class="st">&quot;Various forms of immunotherapy, such as checkpoint blockade immunotherapy, are proving to be effective at restoring T cell-mediated immune responses that can lead to marked and sustained clinical responses, but only in some patients and cancer types. Patients and tumours may respond unpredictably to immunotherapy partly owing to heterogeneity of the immune composition and phenotypic profiles of tumour-infiltrating lymphocytes (TILs) within individual tumours and between patients5,6. Although there is evidence that tumour-mutation-derived neoantigen-specific T cells play a role in tumour control, in most cases the antigen specificities of phenotypically diverse tumour-infiltrating T cells are largely unknown. Here we show that human lung and colorectal cancer CD8+ TILs can not only be specific for tumour antigens (for example, neoantigens), but also recognize a wide range of epitopes unrelated to cancer (such as those from Epstein-Barr virus, human cytomegalovirus or influenza virus). We found that these bystander CD8+ TILs have diverse phenotypes that overlap with tumour-specific cells, but lack CD39 expression. In colorectal and lung tumours, the absence of CD39 in CD8+ TILs defines populations that lack hallmarks of chronic antigen stimulation at the tumour site, supporting their classification as bystanders. Expression of CD39 varied markedly between patients, with some patients having predominantly CD39- CD8+ TILs. Furthermore, frequencies of CD39 expression among CD8+ TILs correlated with several important clinical parameters, such as the mutation status of lung tumour epidermal growth factor receptors. Our results demonstrate that not all tumour-infiltrating T cells are specific for tumour antigens, and suggest that measuring CD39 expression could be a straightforward way to quantify or isolate bystander T cells.&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="anotaciÃ³n-del-experimento" class="section level2">
<h2>AnotaciÃ³n del experimento</h2>
<p>Tenemos que eliminar las subversiones de los genes (el nÃºmero despuÃ©s del punto del nombre en Ensembl de cada gen).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># No funciona sin quitar lo que va despuÃ©s del punto nuestra tabla original</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>tmp<span class="ot">=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">row.names</span>(PRJNA693328))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(PRJNA693328)<span class="ot">=</span>tmp</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>a <span class="ot">=</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db,<span class="at">keys=</span><span class="fu">rownames</span>(PRJNA693328),<span class="at">columns=</span><span class="fu">c</span>(<span class="st">&quot;GENENAME&quot;</span>,<span class="st">&quot;ENTREZID&quot;</span>,<span class="st">&quot;ENSEMBL&quot;</span>,<span class="st">&quot;SYMBOL&quot;</span>),<span class="at">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>b <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">rownames</span>(PRJNA693328),a[,<span class="st">&quot;ENSEMBL&quot;</span>])</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(PRJNA693328) <span class="ot">=</span> a[b,]</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(PRJNA693328)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>PRJNA693328 <span class="ot">=</span> PRJNA693328[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">rowData</span>(PRJNA693328)[,<span class="st">&quot;ENSEMBL&quot;</span>])),]</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>PRJNA693328</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>sel <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">unique</span>(<span class="fu">rowData</span>(PRJNA693328)[,<span class="st">&quot;ENSEMBL&quot;</span>]),<span class="fu">rowData</span>(PRJNA693328)[,<span class="st">&quot;ENSEMBL&quot;</span>])</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>PRJNA693328 <span class="ot">=</span> PRJNA693328[sel,]</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Eliminamos los genes que no tienen ninguna lectura alineada.</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="co">#PodrÃ­amos utilizar criterios mÃ¡s restrictivos en posteriores pasos del anÃ¡lisis.</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>nullsum <span class="ot">=</span> <span class="fu">apply</span>(<span class="fu">assay</span>(PRJNA693328),<span class="dv">1</span>,sum)<span class="sc">==</span><span class="dv">0</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>nullsum</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>PRJNA693328 <span class="ot">=</span> PRJNA693328[<span class="sc">!</span>nullsum,]</span></code></pre></div>
</div>
<div id="limpieza-final" class="section level2">
<h2>Limpieza final</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> Genome <span class="pp">*</span>.gtf <span class="pp">*</span>.bam</span></code></pre></div>
</div>
<div id="guardamos-el-dataset" class="section level2">
<h2>Guardamos el dataset</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(PRJNA693328,<span class="at">file=</span><span class="st">&quot;PRJNA693328.rda&quot;</span>)</span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
